core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1039,7 +1039,15 @@ private Object readResolve() {
         agentProtocols = null;
         return this;
     }
-    
+
+    public ProxyConfiguration getProxy() {
+        return proxy;
+    }
+
+    public void setProxy(ProxyConfiguration proxy) {
```

Comment:
These methods need to be documented at least. You modify the central Jenkins class. 
`proxy` settings also won't be propagated to the disk for this method, a custom configurator would be needed.